### PR TITLE
Update ui npm dependencies to latest versions

### PR DESCRIPTION
**Pull request by saswatamcode**
### Pertaining issues
#3811

### Changes:
- Upgraded  node dependencies of react UI (`pkg/ui/react-app`) to latest releases as much as possible

### Caveats
- Cannot upgrade to `react v17`, since `enzyme` does not have any official adapter for v17. An [unofficial adapter](https://www.npmjs.com/package/@wojtekmaj/enzyme-adapter-react-17) does exist but with [issues](https://github.com/enzymejs/enzyme/issues/2429)
- Upgrading `enzyme`, `jest-fetch-mock` and `typescript` causes certain tests to fail
- Cannot upgrade `eslint` and `react-scripts` to their latest releases without upgrading to `react v17` first, since that may lead to further issues

### CODE REVIEW

**Review by prmsrswt — COMMENTED** [review]
A lot of the errors in the CI are because of bad formatting as we updated prettier which comes with a few different formatting rules. Those can be fixed easily by running `make react-app-lint-fix` in the project root.

A few other failures are due to the lack of typeScript types, you might want to look into them as well.

File: pkg/ui/react-app/src/pages/targets/__testdata__/testdata.ts [hunk 1]
```diff
@@ -1,5 +1,3 @@
-/* eslint @typescript-eslint/camelcase: 0 */
```
Comment (prmsrswt) [inline pkg/ui/react-app/src/pages/targets/__testdata__/testdata.ts:1]: Let's make this `/* eslint-disable camelcase */` and make this rule "warn" for all other files in eslintrc as said in the previous comment.

**Review by prmsrswt — COMMENTED** [review]
Great work! Thanks for taking this up.
I have a few more suggestions and then we can get this merged :muscle:

File: pkg/ui/react-app/.eslintrc.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "plugin:prettier/recommended"
   ],
   "rules": {
-    "@typescript-eslint/camelcase": "warn",
+    "camelcase": 0,
```
[the remark this answers is not in the document]
Comment (saswatamcode) [reply pkg/ui/react-app/.eslintrc.json:9]: Sure!

**Review by prmsrswt — APPROVED** [review]
Awesome work, LGTM!

**Review by squat — APPROVED** [review]
Nice

**Pull request merged**